app/cdap/components/AddTransformation/index.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Button, Container } from '@material-ui/core';
+import DrawerWidget from 'components/DrawerWidget';
+import T from 'i18n-react';
+import React, { Fragment, useState, useEffect } from 'react';
+import SelectColumnsList from 'components/AddTransformation/SelectColumnsList';
+import { useStyles } from 'components/AddTransformation/styles';
+import {
+  IAddTransformationProps,
+  IHeaderNamesList,
+  IMultipleSelectedFunctionDetail,
+  IDataQualityItem,
+} from 'components/AddTransformation/types';
+import { getDataQuality } from 'components/AddTransformation/CircularProgressBar/utils';
+import { multipleColumnSelected, ADD_TRANSFORMATION_PREFIX } from 'components/AddTransformation/constants';
+
+export default function({
+  directiveFunctionSupportedDataType,
```

Comment:
rename plz, dont use directive keyword when working with transformations
also rename to specific feature relatives for better readability